marquez_airflow/dag.py

```diff
@@ -117,74 +117,87 @@ def create_dagrun(self, *args, **kwargs):
 
     def _get_extractor(self, task):
         extractor = self._extractors.get(task.__class__)
-        log.info(f'extractor for {task.__class__} is {extractor}')
+        log.debug(f'extractor for {task.__class__} is {extractor}')
         return extractor
 
     def _timed_log_message(self, start_time):
         return f'airflow_dag_id={self.dag_id} ' \
             f'marquez_namespace={self.marquez_namespace} ' \
             f'duration_ms={(self._now_ms() - start_time)}'
 
+    def _handle_task_instance_state(self, ti, marquez_job_run_id):
+        if ti.state in {State.SUCCESS, State.SKIPPED}:
+            self.get_or_create_marquez_client(). \
+                mark_job_run_as_completed(run_id=marquez_job_run_id)
+        else:
+            self.get_or_create_marquez_client().mark_job_run_as_failed(
+                run_id=marquez_job_run_id)
+
     def handle_callback(self, *args, **kwargs):
         self.log.debug(f"handle_callback({args}, {kwargs})")
 
         try:
             dagrun = args[0]
             self.log.debug(f"handle_callback() dagrun : {dagrun}")
             task_instances = dagrun.get_task_instances()
-            self.log.info(f"{task_instances}")
+
             for ti in task_instances:
-                task = dagrun.get_dag().get_task(ti.task_id)
-                self.log.info(f"ti: {ti} of task: {task}")
+                ti.task = self.get_task(ti.task_id)
+                task = ti.task
```

Comment:
Do we want to modify `ti` here? Is this used somewhere else? Is there a risk to collide with a field in TaskInstance?

Reply:
airflow uses this under the hood. removing to not depend on it.